### ISSUE TITLE
ssbc: Going back loses access to execution

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#37076]
Going back in the browser after clicking "Run execution" navigates back to `/edit`. I thought it would get me back to the editor, but not get rid of the execution in the context. I think the `/edit` page should also load the latest batch spec and populate the page, just like with the direct link. (Or just a history.replace?)

**Comment by courier-new** [comment]
So I agree this is weird/a problem, but I'm thinking of a slightly different solution. I totally agree using `history.replace` makes more sense here when you click "run batch spec", but then separately, if the user lands back on `/edit` by other means, I think we should show a warning banner that there's an active execution and prompt them to view that instead.

If we continued to populate `/edit` with the latest active execution while the user is actively editing their spec, we create the possibility that you have one spec loaded in your editor on the spec tab and a different spec loaded with workspace progress on the execution tab. That's rather disorienting. I remember talking about this on a couple occasions with Rob, and we agreed that if both the "spec" and "execution" tabs are accessible, they should always refer to the same batch spec.

That's also one of the main reasons for the separate `/executions/:id` URL namespace with `/executions/:id/spec` and `/executions/:id/configuration` for the read-only step 1 and step 2. In our current model where we don't really think about parallel executions, leaving this namespace is how we signify the choice to *go back* to step 2 to create a new batch spec to execute something different:
- If you decide you want cancel or edit your batch spec, we take you *back* to `/edit`
- If you just want to view the batch spec you're currently executing, we take you to `/executions/:id/spec`, but you're still "on step 3."

Anyway, my explanation is probably confusing and not the easiest to follow, so I just implemented what I'm suggesting instead. 😅 Check out https://github.com/sourcegraph/sourcegraph/pull/37185!
